capstone-app/src/views/ProfilePage/Sections/SurveyResponse/SurveyResponseSummary.js

```diff
@@ -0,0 +1,162 @@
+  JITTERY: {mainColor: 'rgba(158, 158, 158, 1)', bgColor: 'rgba(238, 238, 238, 1)'},
+  ACTIVE: {mainColor: 'rgba(96, 125, 139, 1)', bgColor: 'rgba(176, 190, 197, 1)'},
+  AFRAID: {mainColor: 'rgba(33, 33, 33, 1)', bgColor: 'rgba(97, 97, 97, 1)'},
+};
+
+/** 
+ * Component for the summary graph, which compares the user's feelings against world feelings 
+ * over time. Includes buttons to display each of the 20 PANAS feelings.
+ */
+export default class SurveyResponseSummary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loaded: [],
+      data: {
+				datasets: []
+			}
+    };
+  }
+
+  /** 
+  * Makes a GET request for all survey responses containing the specified feeling, then passes the 
+  * responses along to be sorted.
+  */
+  async handleClick(feeling, event) {
+    if (!this.state.loaded.includes(feeling)) {
+      var newLoaded = this.state.loaded.concat(feeling);
+      this.setState({ loaded: newLoaded });
+
+      var requestParam = new URLSearchParams();
+      requestParam.append('action', 'feeling');
+      requestParam.append('data', feeling);
+      
+      const request = new Request(
+        'https://manage-at-scale-step-2020.appspot.com/survey?' + requestParam
+      );
+
+      let responses = await fetch(request).then(response => response.json());
+
+      this.sortResponsesByUser(responses, feeling);
+    }
+  }
+
+  /** 
+  * Sorts responses by user and adds two new datasets to the graph - one representing how the user 
+  * has logged the specified feeling over time, and the other representing how the world has done so.
+  */
+  sortResponsesByUser(responses, feeling) {
+    var userResponses = [];
+    var worldResponses = [];
+    for (const response of responses) {
+      if (response.user === this.props.user) {
+        userResponses.push(response);
+      } else {
+        worldResponses.push(response);
+      }
+    }
+    var newDatasets = this.state.data.datasets;
+    newDatasets.push(this.buildDataset(userResponses, true, feeling));
+    newDatasets.push(this.buildDataset(worldResponses, false, feeling));
+
+    var newData = {datasets: newDatasets};
+    this.setState({ data: newData });
+  }
+
+  /** 
+  * Builds a dataset of (x,y) values from the given responses, where the x-axis is time 
+  * and the y-axis is the intensity of the specified feeling.
+  */
+  buildDataset(responses, isUserData, feeling) {
+    var sortedResponses = responses.sort(this.compareTimestamp);
+    const dataset = {
+      fill: false,
+      data: []
+    }
+
+    if (isUserData) {
+      dataset.label = feeling + ' (you)';
+      dataset.backgroundColor = feelings[feeling].mainColor;
+      dataset.borderColor = feelings[feeling].mainColor;
+    } else {
+      dataset.label = feeling + ' (world)';
+      dataset.backgroundColor = feelings[feeling].bgColor;
+      dataset.borderColor = feelings[feeling].bgColor;
+    }
+
+    var intensities = {
```

Comment:
I see the same var defined in another file, is there a way to reuse it?

Reply:
Yes! Moving it to the parent component and passing it in as a prop - 76e31e6152721fae7cca2d3980456ee17f791772.